pkg/apis/serving/v1alpha1/service_validation.go

```diff
@@ -83,10 +85,36 @@ func (ss *ServiceSpec) Validate(ctx context.Context) *apis.FieldError {
 		errs = errs.Also(ss.DeprecatedPinned.Validate(ctx).ViaField("pinned"))
 	}
 
+	// Before checking ConfigurationSpec, check RouteSpec.
+	if len(set) > 0 && len(ss.RouteSpec.Traffic) > 0 {
+		errs = errs.Also(&apis.FieldError{
```

Comment:
This seems like we could use the MultipleOneOf instead of a new error type.

Reply:
Done